kolibri/core/assets/src/api_resource.js

```diff
@@ -158,18 +165,29 @@ class Resource {
    */
   constructor(kolibri) {
     this.models = {};
+    this.collections = {};
     this.kolibri = kolibri;
   }
 
   /**
    * Optionally pass in data and instantiate a collection for saving that data or fetching
    * data from the resource.
+   * @param {Object} params - default parameters to use for Collection fetching.
    * @param {Object[]} data - Data to instantiate the Collection - see Model constructor for
    * details of data.
    * @returns {Collection} - Returns an instantiated Collection object.
    */
-  getCollection(data = []) {
-    return new Collection(data, this);
+  getCollection(params = {}, data = []) {
+    let collection;
+    // Note: to keep the key generation cheap, caching will only work for objects constructed
+    // with the same key order.
+    const key = JSON.stringify(params);
```

Comment:
This will be different if the params get passed in a different order. Without an expensive deep comparison, this works well enough.